packages/server/graphql/public/rules/isViewerOnTeam.ts

```diff
@@ -6,15 +6,17 @@ import {GQLContext} from '../../graphql'
 type GetTeamId = (source: any, args: any, context: GQLContext) => Promise<string | Error>
 
 const isViewerOnTeam = (getTeamId: GetTeamId) =>
-  rule({cache: 'contextual'})(async (source, args, context: GQLContext) => {
-    const {authToken, dataLoader} = context
-    const viewerId = getUserId(authToken)
-    const teamId = await getTeamId(source, args, context)
-    if (teamId instanceof Error) return teamId
-    const teamMemberId = TeamMemberId.join(teamId, viewerId)
-    const teamMember = await dataLoader.get('teamMembers').load(teamMemberId)
-    if (!teamMember) return new Error('Viewer in not on team')
-    return true
-  })
+  rule(`isViewerOnTeam-${getTeamId}`, {cache: 'contextual'})(
```

Comment:
now that i read this again, since getTeamId _may_ also use `source, args`, should we change this cache to `strict`? 

+1 since `getTeamId` is a function, and we're putting it in a string, JS is gonna call `getTeamId.toString()`, which will result in the body of the function, right? should it instead be `getTeamId.name`?
is it enough to use the name of the `getTeamId` function, or do we need to somehow calculate the teamId?

Reply:
We should switch to strict, you're right.
`.name` should be enough unless we're also passing anonymous functions at times, then it would lead to conflicts, so serializing the body is safer.